### ISSUE TITLE
Direct casting does not become `global::` form

### ISSUE THREAD

**Issue by lsoft** [zompinc/sync-method-generator#27]
create console app and put the following:

```C#
using CustomNamespace;
using System.Net.NetworkInformation;
using System.Net.Sockets;
using Zomp.SyncMethodGenerator;

namespace SomeNamespace
{
    internal class Program
    {
        static void Main(string[] args)
        {
            Console.WriteLine("Hello, World!");
        }
    }

    public partial class Subject
    {
        [CreateSyncVersion]
        public async Task<object> SomeMethodAsync(object o)
        {
            await Task.Delay(1);

            return (CustomClass)o;
        }

    }
}

namespace CustomNamespace
{
    public class CustomClass { }
}
```
produced file will be:

```C#
// <auto-generated/>
#nullable enable
namespace SomeNamespace
{
    public partial class Subject
    {
        public object SomeMethod(object o)
        {

            return (CustomClass)o;         <-- here is error, CustomClass must be in global-form, because of absense of any using
        }
    }
}
```

Looks like it is the showstopper for me, I have to stop in switching my project from sync style to sync+async style. Sorry to issue-spam!

**Comment by lsoft** [comment]
why not to copy all the usings into generated files and NOT to convert any type to `global` form?

**Comment by virzak** [comment]
Thanks for the issue spam. Will resolve soon.

As for usings, see this: https://github.com/zompinc/sync-method-generator/issues/13#issuecomment-1809214590

**Comment by virzak** [comment]
All the recent issues are fixed in version 1.2.8, now available on nuget. Thanks for the ⭐ and the interest.
С новым годом, с новым счастьем!
